state-sync/state-sync-v1/src/request_manager.rs

```diff
@@ -243,8 +243,10 @@ impl RequestManager {
             // At minimum, go through networks with preference level <= multicast level.
             // If no peers are found for the current multicast level, continue doing
             // best effort search of the networks to failover to.
-            if !chosen_peers.is_empty() && *network_level >= self.multicast_network_level {
-                new_multicast_network_level = Some(network_level);
+            if !chosen_peers.is_empty() {
```

Comment:
Maybe I've been doing too much day drinking but how is this code logically different? 😂

Reply:
I looked at it closer, it's cause you have to look at the whole loop.

When the network level is upgraded, it sends to only one node.  Otherwise, it sends to all nodes in a network level.

Reply:
but ... but ... there's something quirky here, let's pause